sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamSqlEnv.java

```diff
@@ -49,19 +46,19 @@
 @Experimental
 public class BeamSqlEnv {
   final JdbcConnection connection;
-  final BeamQueryPlanner planner;
+  final QueryPlanner planner;
 
-  private BeamSqlEnv(TableProvider tableProvider) {
-    connection = JdbcDriver.connect(tableProvider);
-    planner = new BeamQueryPlanner(connection);
+  private BeamSqlEnv(JdbcConnection connection, QueryPlanner planner) {
+    this.connection = connection;
+    this.planner = planner;
   }
 
   public static BeamSqlEnv readOnly(String tableType, Map<String, BeamSqlTable> tables) {
     return withTableProvider(new ReadOnlyTableProvider(tableType, tables));
   }
 
   public static BeamSqlEnv withTableProvider(TableProvider tableProvider) {
-    return new BeamSqlEnv(tableProvider);
+    return BeamSqlEnvBuilder.builder(tableProvider).build();
```

Comment:
nit:I think the pattern is to have a `static BeamSqlEnv.builder()` method that would call a protected constructor on the builder